From dbce56d62584adf30f5bda190fcfba86738f587c Mon Sep 17 00:00:00 2001
From: Ran Shalit <rshalit@habana.ai>
Date: Wed, 18 Sep 2024 18:16:02 +0300
Subject: [PATCH] [FSW-1160] package: dsm: use run_from_iccm in build

Running dsm from ICCM is convenient in palladium, but
with the general flow from flash it will not run from ICCM.
Hence, for running from ICCM it we use variable in build:

make fs1_arc-pldm-prepare;make dsm-dirclean dsm-rebuild;
make fs1-pldm-build DSM_RUN_FROM_ICCM=true

Change-Id: I0a4185455c53be819b4e64c5f1351d2f60033421
Signed-off-by: Ran Shalit <rshalit@habana.ai>
---

diff --git a/package/dsm/dsm.mk b/package/dsm/dsm.mk
index 76ccc4e..bdf98b0 100644
--- a/package/dsm/dsm.mk
+++ b/package/dsm/dsm.mk
@@ -10,6 +10,10 @@
 DSM_APP_VERSION = $(shell cut -d- -f2- $(@D)/localversion-habanalabs)
 DSM_LOCAL_VERSION = $$(sed "s/^.*fw-\\([0-9]\\+\\.[0-9]\\+\\.[0-9]\\+\\).*$$/\\1/" $(@D)/localversion-habanalabs)
 
+ifeq ($(DSM_RUN_FROM_ICCM),true)
+DSM_CONF_APP_OPTS += -DCONFIG_DSM_RUN_FROM_ICCM=y
+endif #PREBOOT_RUN_FROM_ICCM
+
 DSM_BUILD_DIR = $(@D)
 DSM_ZEPHYR_BUILD_DIR ?= $(@D)/zephyr
 
@@ -52,6 +56,7 @@
 
 DSM_CONF_OPTS  = $(ZEPHYR_CONF_OPTS)
 DSM_CONF_OPTS += -UCMAKE_TOOLCHAIN_FILE
+DSM_CONF_OPTS += $(DSM_CONF_APP_OPTS)
 DSM_CONF_OPTS += -DUSER_CACHE_DIR=$(@D)/.cache
 DSM_CONF_OPTS += -DCMAKE_EXE_LINKER_FLAGS_INIT=-Wl,-e__start
 DSM_CONF_OPTS += -DEXTRA_CFLAGS="\
